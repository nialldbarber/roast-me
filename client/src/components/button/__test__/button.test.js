import React from 'react'
import '@testing-library/jest-dom/extend-expect'
import renderer from 'react-test-renderer'
import Button from '~@components/button'

describe('<Button />', () => {
  it('renders button text', () => {
    const component = renderer.create(<Button text="Button text" />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders button id when passed', () => {
    const component = renderer.create(<Button id="button-id" />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('matches snapshot', () => {
    const message = 'Button text'
    const tree = renderer.create(<Button text={message} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
