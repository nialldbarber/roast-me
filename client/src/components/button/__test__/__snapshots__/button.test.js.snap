// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> matches snapshot 1`] = `
<button
  className="sc-bdVaJa fyGyMf btn"
  type="submit"
>
  <span>
    <span>
      Button text
    </span>
    
  </span>
</button>
`;

exports[`<Button /> renders button id when passed 1`] = `
<button
  className="sc-bdVaJa fyGyMf btn"
  id="button-id"
  type="submit"
>
  <span>
    <span />
    
  </span>
</button>
`;

exports[`<Button /> renders button text 1`] = `
<button
  className="sc-bdVaJa fyGyMf btn"
  type="submit"
>
  <span>
    <span>
      Button text
    </span>
    
  </span>
</button>
`;
